cohortextractor/tpp_backend.py

```diff
@@ -848,6 +848,36 @@ def patients_registered_practice_as_of(self, date, returning=None):
             """,
         )
 
+    def patients_date_deregistered_from_all_supported_practices(self, between):
+        if between is None:
+            between = (None, None)
+        min_date, max_date = between
+        # Registrations with no end date are recorded using an end date of
+        # 9999-12-31, so if no max date is supplied then we need to set a max
+        # date to something less than 9999-12-31 to filter out registrations
+        # which have no end date.
+        if max_date is None:
+            max_date = "3000-01-01"
```

Comment:
Giving the Y3K consultants something to do?

Reply:
64KB ought to be enough for anybody...